superset/db_engine_specs/elasticsearch.py

```diff
@@ -41,7 +41,7 @@ class ElasticSearchEngineSpec(BaseEngineSpec):
     type_code_map: Dict[int, str] = {}  # loaded from get_datatype only if needed
 
     @classmethod
-    def convert_dttm(cls, target_type: str, dttm: datetime) -> str:
-        if target_type.upper() in ("DATETIME", "DATE"):
-            return f"'{dttm.isoformat()}'"
-        return f"'{dttm.strftime('%Y-%m-%d %H:%M:%S')}'"
+    def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
+        if target_type.upper() in ("DATE", "DATETIME"):
+            return f"CAST('{dttm.isoformat()}' AS TIMESTAMP)"
```

Comment:
@dpgaspar I know you recently added this and I was hoping you could validate the logic.